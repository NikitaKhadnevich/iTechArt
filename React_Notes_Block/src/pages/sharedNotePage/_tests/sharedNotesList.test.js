/* eslint-disable no-shadow */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-unused-vars */
import React from 'react';
import * as reactRedux from 'react-redux';
import { Provider } from 'react-redux';
import { configure, shallow, mount, create, render } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import configureMockStore from 'redux-mock-store'; // ES6 modules
import { BrowserRouter as Router } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from 'react-query';
import SharedNotesList from '../sharedNotesList';

configure({ adapter: new Adapter() });

describe('Shared Notes', () => {
  const middlewares = [];
  const mockStore = configureMockStore(middlewares);
  const initialState = {
    userData: [
      {
        id: '3',
        title: 'Some Title',
        description: 'Some describtion',
        date: '10/19/2021, 04:19:12 AM',
        sharedTime: '',
      },
    ],
  };
  const store = mockStore(initialState);
  const sharedList = store.getState().userData;
  const queryClient = new QueryClient();
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

  it('Is it ItemsNavi Exist', () => {
    const SharedWrapper = shallow(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <SharedNotesList />
        </QueryClientProvider>
      </Provider>
    );
    expect(SharedWrapper).toHaveLength(1);
  });
  it('Is it ItemsNavi Snap', () => {
    const SharedWrapper = shallow(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <SharedNotesList />
        </QueryClientProvider>
      </Provider>
    );
    expect(SharedWrapper).toMatchSnapshot();
  });
});
