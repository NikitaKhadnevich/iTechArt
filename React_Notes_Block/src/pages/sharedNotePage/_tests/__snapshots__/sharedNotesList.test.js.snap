// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Notes Is it ItemsNavi Snap 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <QueryClientProvider
    client={
      QueryClient {
        "defaultOptions": Object {},
        "mutationCache": MutationCache {
          "config": Object {},
          "listeners": Array [],
          "mutationId": 0,
          "mutations": Array [],
        },
        "mutationDefaults": Array [],
        "queryCache": QueryCache {
          "config": Object {},
          "listeners": Array [],
          "queries": Array [],
          "queriesMap": Object {},
        },
        "queryDefaults": Array [],
      }
    }
  >
    <SharedNotesList />
  </QueryClientProvider>
</ContextProvider>
`;
